openapi/ui_openapi.yaml

```diff
@@ -835,6 +944,125 @@ components:
       type: string
       # TODO: This could in theory be an enum
 
+
+    SyncList:
+      title: 'SyncList'
+      description: >
+        A SyncList is a list of Collections or Namespaces that will
+        be synced to a Repository.
+      type: object
+      properties:
+        id:
+          title: ID
+          type: integer
+          readOnly: true
+        name:
+          title: Name
+          type: string
+          maxLength: 64
+          minLength: 1
+          example: 'sync_example_com_namespace'
+        policy:
+          $ref: '#/components/schemas/SyncListPolicy'
+        repository:
+          # TODO: may become a repository name or a RepositorySummary
+          title: Repository ID
+          description: 'The ID of the Repository to sync to'
+          type: string
+          format: uuid
+        collections:
```

Comment:
`collections` and `namespaces` list can be eventually large. I think it's better to return and modify them via separate API requests. (e.g. `(POST|DELETE) /api/v1/synclists/{synclist_id}/(collections|namespaces)`)
For example adding \ deleting single namespace to a list should not cause sending back and forth this whole large object.

Reply:
The UI is going to have to load all of the collections in a sync list anyway, so I'm not sure we're saving ourselves any trouble by moving this to a separate API.

Reply:
I would like to add a new parameter on the namespace and collection list APIs that allows for the UI to query collections/namespaces by synclist id

Reply:
@newswangerd That sounds reasonable at first glance. I'll consider that a different issue/pr though since it is change to namespace viewsets and collection viewsets (which may potentially need tweaks in pulpcore / pulp_ansible)

Reply:
This might be a good opportunity to look into migrating the ui collection viewset away from proxying to pulp.